{"CVE Code": "CVE-2022-41940", "Description": "Engine.IO is the implementation of transport-based cross-browser/cross-device bi-directional communication layer for Socket.IO. A specially crafted HTTP request can trigger an uncaught exception on the Engine.IO server, thus killing the Node.js process. This impacts all the users of the engine.io package, including those who uses depending packages like socket.io. There is no known workaround except upgrading to a safe version. There are patches for this issue released in versions 3.6.1 and 6.2.1.", "NVD Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-41940", "Vulnerability Categories": "N/A", "Affected Products": "\nSocket\u00bbEngine.io\u00bbForNode.jsVersions  from including (>=) 4.0.0andbefore (<) 6.2.1cpe:2.3:a:socket:engine.io:*:*:*:*:*:node.js:*:*Matching versions\nSocket\u00bbEngine.io\u00bbForNode.jsVersions  before (<) 3.6.1cpe:2.3:a:socket:engine.io:*:*:*:*:*:node.js:*:*Matching versions", "EPSS Score": "0.07% Probability of exploitation activity in the next 30 days\n~ 34 % Percentile, the proportion of vulnerabilities that are scored at or less", "EPSS History": "\n #  Date  Old EPSS Score  New EPSS Score  Delta (New - Old) \n 1  2024-12-17  0.10%  0.07%  -0.02 \n 2  2024-07-13  0.08%  0.10%  +0.01 \n 3  2024-06-14  0.08%  0.08%  +0.00 \n 4  2023-11-28  0.07%  0.08%  +0.01 \n 5  2023-11-17  0.07%  0.07%  0.00 \n 6  2023-09-05  0.07%  0.07%  +0.00 \n 7  2023-03-07  0.89%  0.07%  -0.82 ", "CVSS Scores": "6.5", "CWE": [{"CWE_ID": "CWE-248 Uncaught Exception", "CWE_link": "/cwe-details/248/Uncaught-Exception.html"}], "Reference": [{"ref_link": "https://github.com/socketio/engine.io/commit/83c4071af871fc188298d7d591e95670bf9f9085", "ref_desc": "fix: catch errors when destroying invalid upgrades \u00b7 socketio/engine.io@83c4071 \u00b7 GitHubPatch;Third Party Advisory", "ref_cve_link": "N/A"}, {"ref_link": "https://github.com/socketio/engine.io/security/advisories/GHSA-r7qp-cfhv-p84w", "ref_desc": "Uncaught exception in engine.io \u00b7 Advisory \u00b7 socketio/engine.io \u00b7 GitHubExploit;Third Party Advisory", "ref_cve_link": "N/A"}, {"ref_link": "https://github.com/socketio/engine.io/commit/425e833ab13373edf1dd5a0706f07100db14e3c6", "ref_desc": "fix: catch errors when destroying invalid upgrades (#658) \u00b7 socketio/engine.io@425e833 \u00b7 GitHubPatch;Third Party Advisory", "ref_cve_link": "N/A"}]}